src/plugins/pcr/src/InSilicoPcrOptionPanelWidget.cpp

```diff
@@ -128,14 +128,19 @@ void InSilicoPcrOptionPanelWidget::sl_onPrimerChanged() {
     }
 
     QString message;
-    bool correct = PrimerStatistics::checkPcrPrimersPair(forward, reverse, message);
+    bool isCriticalError;
+    bool correct = PrimerStatistics::checkPcrPrimersPair(forward, reverse, message, isCriticalError);
     if (correct) {
         warningLabel->hide();
         findProductButton->setText(tr("Find product(s)"));
+        detailsLinkLabel->show();
     } else {
         warningLabel->show();
         warningLabel->setText(tr("Warning: ") + message);
         findProductButton->setText(tr("Find product(s) anyway"));
+        if (isCriticalError) {
+            detailsLinkLabel->hide();
```

Comment:
A1) if isCorrect = true => the details are shown.
A2) If next state changes to isCorrect = false and critical error => details are hidden
A3) If next state is isCorrect & no critical error => details stays hidden.

but
B1) if isCorrect = true => the details are shown.
B2) If next state changes to isCorrect = false and no critical error => details are shown

B2 is the same state with A3, but the state of the detailsLinkLabel is different?
Question